Added Koashur

FarhanSyedain opened the pull request:
This PR adds the Koashur Khat layout to the Futo keyboard project. Koashur Khat is a Roman-script representation of the Kashmiri language, designed for easy typing on standard keyboards.

Features:

Full Koashur Khat character set with vowels and special diacritics.
Supports future dead keys for mapping input sequences to Koashur Khat symbols.
Long-press options for additional characters where applicable.
Fully compatible with Futo’s existing layout framework.
This addition enables efficient typing of Kashmiri text while preserving all phonetic nuances of the Koashur Khat system.
Adding the Koashur Khat layout to Futo is an important step toward making digital communication accessible for Kashmiri speakers. This layout provides a standardized, phonetic way to type Kashmiri on a regular keyboard, eliminating the need for external tools or workarounds. By supporting Koashur Khat, Futo not only aids in the preservation and daily use of the Kashmiri language but also empowers students, researchers, and language enthusiasts to write accurately and efficiently. Including this layout enhances Futo’s inclusivity and ensures that Kashmiri users can participate fully in digital communication, education, and content creation.

FarhanSyedain committed:
Added Koashur

abb128 commented:
`ˇ + e = ë`
I wanna say dead keys or combining keys shouldn't work like this, a caron shouldn't produce a diaeresis, do you think having these keys accessible via long-press instead would be suitable?

FarhanSyedain commented:
No not actually since then there would be no need to add this pr anyway. The diacritics are used a lot and therefore must be easily accessible.I have taken inspiration from the Czech keyboard on iOS which does the same thing with dead carron and '

tenextractor commented:
@FarhanSyedain

> The diacritics are used a lot and therefore must be easily accessible.I have taken inspiration from the Czech keyboard on iOS which does the same thing with dead carron and '
ď is actually `U+010F LATIN SMALL LETTER D WITH CARON`. The caron is just displayed like an apostrophe in printed fonts.

The system you propose (ˇ + e -> ë) is not possible with the current backend. For a proper implementation of such a system with the best usability, we would have to implement a layer system such that, when the user presses on the ˇ key, the whole keyboard shifts into a different layer like this:
```

FarhanSyedain commented:
> @FarhanSyedain
> 
> > The diacritics are used a lot and therefore must be easily accessible.I have taken inspiration from the Czech keyboard on iOS which does the same thing with dead carron and '
> 
> ď is actually `U+010F LATIN SMALL LETTER D WITH CARON`. The caron is just displayed like an apostrophe in printed fonts.
> 
> The system you propose (ˇ + e -> ë) is not possible with the current backend. For a proper implementation of such a system with the best usability, we would have to implement a layer system such that, when the user presses on the ˇ key, the whole keyboard shifts into a different layer like this:
> 
> ```
> q ě ë ē t ÿ u i ö p
> a s d f g h j k l
> z x c v b n m
> ```
> 
> There is already an altPages mechanism that is similar, but does not work for scripts with uppercase and lowercase.
> 
> I also disagree on making ˇ + r -> ě and ˇ + e -> ë because that is not intuitive.

For the backend, I updated and added a custom deadkeycombiner just like the Korean combiner. It supports the proposed layout. 

I agree with some points and for that I propose long pressing e to get the rest of the least used characters like e with two dots and e with a line on top. Since, y can be used and is standard for non ascii usage in koashurkhat. Please check the updated backend

FarhanSyedain commented:
@tenextractor also check pr on andriod-keyboard

FarhanSyedain committed:
Remoed useless comments

FarhanSyedain commented:
@tenextractor @abb128 @Zvonimir-FUTO please check the andriod keyboard pr [https://github.com/futo-org/android-keyboard/pull/1579](url) which complements the current pr and adds backend mapping and custom dead key handler

FarhanSyedain commented:
No other layouts do the same. Thanks for adding.

Youripaste commented:
They wont tell you when is next release, they dont care about users nor programmers, they recieve many money, but only have one dev: Alex. Here is only mantainers that anwser when they are bored. Also, you are a dreamer, do you realy think your invented keyboard, that is not standard nor user by anyone is realy be aproved? Your language uses persian script by law. Not latin script. The proposed romanization you made yourself is not aproved by anyone, and its very different from the homemade ones. Are you a linguist? What autorithy you have to decide the best romanization for a language for al people? No one types Koashur in Latin, its only "proposals" its not realistic to expect your keyboard or be added to a great app like FUTO. They are afraid to reject you, im just telling you the true. It wont be aproved. They dont care even paid users! Nor take critics well. See > "_Obscure_ layouts can serve to confuse users and bloat the layout list, so we are likely going to have some standards for inclusion of layouts." If no other keyboard uses it it will 100% be rejected! Koashurkhat on redit has no post and 0 users. **FUTO kills the language everyday!!** 

Where is the Koashur layout? FarhanSyedain langugae is dieying, its urgent to be merged! RN! FarhanSyedain is the savior or Kshmir romanization homemade no susers. FUTO team disaponts. and dont care our cashmirrow community!!!

FarhanSyedain commented:
@Zvonimir-FUTO @tenextractor i got some haters and spoamers up my ass

FarhanSyedain committed:
Merge branch 'futo-org:main' into main
